command/init.go

```diff
@@ -77,31 +77,48 @@ func initCommand(cctx *cli.Context) error {
 
 	adminAddr := cctx.String("listen-admin")
 	if adminAddr != "" {
-		_, err := multiaddr.NewMultiaddr(adminAddr)
-		if err != nil {
-			return fmt.Errorf("bad listen-admin: %s", err)
+		if adminAddr != "none" {
+			_, err := multiaddr.NewMultiaddr(adminAddr)
+			if err != nil {
+				return fmt.Errorf("bad listen-admin: %s", err)
+			}
 		}
 		cfg.Addresses.Admin = adminAddr
 	}
 
 	finderAddr := cctx.String("listen-finder")
 	if finderAddr != "" {
-		_, err := multiaddr.NewMultiaddr(finderAddr)
-		if err != nil {
-			return fmt.Errorf("bad listen-finder: %s", err)
+		if finderAddr != "none" {
+			_, err := multiaddr.NewMultiaddr(finderAddr)
+			if err != nil {
+				return fmt.Errorf("bad listen-finder: %s", err)
+			}
 		}
 		cfg.Addresses.Finder = finderAddr
 	}
 
 	ingestAddr := cctx.String("listen-ingest")
 	if ingestAddr != "" {
-		_, err := multiaddr.NewMultiaddr(ingestAddr)
-		if err != nil {
-			return fmt.Errorf("bad listen-ingest: %s", err)
+		if ingestAddr != "none" {
+			_, err := multiaddr.NewMultiaddr(ingestAddr)
+			if err != nil {
+				return fmt.Errorf("bad listen-ingest: %s", err)
+			}
 		}
 		cfg.Addresses.Ingest = ingestAddr
 	}
 
+	p2pAddr := cctx.String("listen-p2p")
+	if p2pAddr != "" {
```

Comment:
should this check be `if cctx.IsSet("listen-p2p")`?

Reply:
No, because doing:
```shell
./storetheindex init --listen-p2p=""
```
results in `cctx.IsSet` returning `true`, but the string being empty `""`